### PR TITLE
Ops 682/task 1358 pagination

### PR DESCRIPTION
**Pull request by johndeange**
## What changed

Create PaginationNav component and add to BLI List page.

## Issue

https://github.com/HHS/OPRE-OPS/issues/1358

## How to test

Visit the BLI List page and interact with the pagination component there.

## Screenshots

image

### CODE REVIEW

File: frontend/src/components/UI/AllBudgetLinesTable/AllBudgetLinesTable.js [hunk 1]
```diff
@@ -13,6 +16,13 @@ import { All_BUDGET_LINES_TABLE_HEADINGS } from "../../../constants";
  * @returns {React.JSX.Element} The TableRow component.
  */
 const AllBudgetLinesTable = ({ budgetLines }) => {
+    const [currentPage, setCurrentPage] = useState(1);
+
+    const BLIS_PER_PAGE = 10;
```
Comment (fpigeonjr) [inline frontend/src/components/UI/AllBudgetLinesTable/AllBudgetLinesTable.js:21]: should this live in the `constants.js` file?

File: frontend/src/components/UI/PaginationNav/PaginationNav.jsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+/**
+ * The PaginationNav component provides pagination for a table
+ * ref: https://designsystem.digital.gov/components/pagination/
+ * @param {object} props - The component props.
+ * @param {number} [props.currentPage] - The current page number.
+ * @param {Function} [props.setCurrentPage] - A function to call to set the current page.
+ * @param {object[]} [props.items] - An array of objects to paginate over.
+ * @param {number} [props.itemsPerPage] - The number of items to show per page.
+ * @returns {React.JSX.Element} - The rendered component.
+ * General component properties:
```
Comment (fpigeonjr) [inline frontend/src/components/UI/PaginationNav/PaginationNav.jsx:14]: 🧡 this is really helpful

File: frontend/src/components/UI/PaginationNav/PaginationNav.module.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.hideElement {
```
Comment (fpigeonjr) [inline frontend/src/components/UI/PaginationNav/PaginationNav.module.scss:1]: we may have a this css class in our css globals

**Review by fpigeonjr — APPROVED** [review]
💪 looks great @johndeange. 
Can we get a `cursor-pointer` over the buttons?

File: frontend/src/components/UI/PaginationNav/PaginationNav.module.scss (hunk 3 above)
Comment (fpigeonjr) [reply frontend/src/components/UI/PaginationNav/PaginationNav.module.scss:1]: sorry was thinking about `display-none` from [uswds](https://designsystem.digital.gov/utilities/display/#display-2)

Comment (johndeange) [reply frontend/src/components/UI/PaginationNav/PaginationNav.module.scss:1]: didn't see it anyplace

**Review by stevtek — APPROVED** [review]

**Pull request merged**